File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/sound/PlayerEventsListener.java
/**
 * @file
 * @brief Player events listener interface
 * @author vitamin.caig@gmail.com
 */

package app.zxtune.sound;

import androidx.annotation.NonNull;

/**
 * All calls are synchronous and may be performed from background thread
 */
public interface PlayerEventsListener {

  /**
   * Called when playback is started or resumed
   */
  void onStart();

  /**
   * Called when seek is started
   */
  void onSeeking();

  /**
   * Called when played stream come to an end
   */
  void onFinish();

  /**
   * Called when playback stopped (also called after onFinish)
   */
  void onStop();

  /**
   * Called on unexpected error occurred
   * @param e Exception happened
   */
  void onError(@NonNull Exception e);
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/sound/PlayerEventsListener.kt
/**
 * @file
 * @brief Player events listener interface
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.sound

/**
 * All calls are synchronous and may be performed from background thread
 */
interface PlayerEventsListener {
    /**
     * Called when playback is started or resumed
     */
    fun onStart()

    /**
     * Called when seek is started
     */
    fun onSeeking()

    /**
     * Called when played stream come to an end
     */
    fun onFinish()

    /**
     * Called when playback stopped (also called after onFinish)
     */
    fun onStop()

    /**
     * Called on unexpected error occurred
     * @param e Exception happened
     */
    fun onError(e: Exception)
}
